docs/data/date-pickers/localization/localization.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Pickers - Localization
+---
+
+# Pickers - Localization
+
+<p class="description">Pickers allow to support users from different locales, with formatting, RTL, and localized strings.</p>
+
+The default locale of MUI is English (United States). If you want to use other locales, follow the instructions below.
+
+## Localization
+
+Localization can impact pickers components rendering in two distincts ways: The date format, and the components attributes such as `aria-label`.
+
+### Date-engine locale
+
+Use `LocalizationProvider` to change the date-engine locale that is used to render pickers. Here is an example of changing the locale for the `date-fns` adapter:
+
+{{"demo": "LocalizedDatePicker.js"}}
+
+### Translation keys
+
+As the rest of MUI components, you can modify text and translations.
```

Comment:
Would be great to create a new issue with auto update like for the grid (or re-use the same one)

In another PR of course

Reply:
Issue created #4914